### PR TITLE
feat: few shot + datasets

### PR DESCRIPTION
**Pull request by jakerachleff**
Quick how to guide on using few shot + datasets

### CODE REVIEW

File: versioned_docs/version-2.0/how_to_guides/datasets/index_datasets_for_dynamic_few_shot_example_selection.mdx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  TypeScriptBlock,
+} from "@site/src/components/InstructionsWithCode";
+
+# Dynamic few shot example selection
+
+:::note
+This feature is currently in closed beta. Please sign up [here](https://forms.gle/in9R6t9HNSYMBt7P7) for access
+:::
+
+Configure your datasets so that you can search for few shot examples based on an incoming request.
```
Comment (akira) [inline versioned_docs/version-2.0/how_to_guides/datasets/index_datasets_for_dynamic_few_shot_example_selection.mdx:17]: Do you want to add more background here on why folks would want to use this and what the feature does?

File: versioned_docs/version-2.0/how_to_guides/datasets/index_datasets_for_dynamic_few_shot_example_selection.mdx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+4. You must be on LangSmith cloud
+
+## Index your dataset to be searched
+
+On the datasets UI, click the `Few-shot Index` on the top right corner and hit `Start Sync`.
+
+![](../static/few-shot-index.png)
+
+This process will start to index your data to be searchable in the background. A note will appear on the
+modal above that says if your index is up to date, and if not, what version of your dataset it last indexed.
```
Comment (akira) [inline versioned_docs/version-2.0/how_to_guides/datasets/index_datasets_for_dynamic_few_shot_example_selection.mdx:33]: So would this modal get updated once the background process is done, or need to refresh the page?

File: versioned_docs/version-2.0/how_to_guides/datasets/index_datasets_for_dynamic_few_shot_example_selection.mdx (hunk 1 above)
Comment (jakerachleff) [reply versioned_docs/version-2.0/how_to_guides/datasets/index_datasets_for_dynamic_few_shot_example_selection.mdx:17]: IIUC how to guides are not supposed to do this:

https://diataxis.fr/how-to-guides/#key-principles

> Anything else that’s added distracts both you and the user and dilutes the useful power of the guide. Typically, the temptations are to explain or to provide reference for completeness. Neither of these are part of guiding the user in their work. They get in the way of the action; if they’re important, link to them.

and

>How-to characteristics

> * focused on tasks or problems

> * assume the user knows what they want to achieve

> * action and only action

> * no digression, explanation, teaching

Comment (jakerachleff) [reply versioned_docs/version-2.0/how_to_guides/datasets/index_datasets_for_dynamic_few_shot_example_selection.mdx:17]: We may want a tutorial on few shot prompting that goes through some teaching in addition?

File: versioned_docs/version-2.0/how_to_guides/datasets/index_datasets_for_dynamic_few_shot_example_selection.mdx (hunk 2 above)
Comment (jakerachleff) [reply versioned_docs/version-2.0/how_to_guides/datasets/index_datasets_for_dynamic_few_shot_example_selection.mdx:33]: It polls every 10 seconds:

```
 const { data: syncStatus, error } = useGetSyncStatus(datasetId ?? '', {
    refreshInterval: 10000,
    skip: !fewShotEnabled,
  });
```

File: versioned_docs/version-2.0/how_to_guides/datasets/index_datasets_for_dynamic_few_shot_example_selection.mdx (hunk 1 above)
Comment (akira) [reply versioned_docs/version-2.0/how_to_guides/datasets/index_datasets_for_dynamic_few_shot_example_selection.mdx:17]: Yeah, that makes sense on tutorial

**Review by akira — APPROVED** [review]

**Pull request merged**